shows top-3 and comprehensive list of movies in any genre

Koala-t opened the pull request:
handle missing posters if there's time.

Koala-t committed:
shows top-3 and comprehensive list of movies in any genre

salogel42 commented on app/helpers/movies_helper.rb:
```diff
@@ -3,4 +3,16 @@ def get_gif(genre)
+    Movie.all.each do |movie|
+      movies << movie if movie.genres.include?(genre)
+    end
+    movies
```
use `select` or `find_all` http://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-select

salogel42 commented on app/helpers/movies_helper.rb:
```diff
@@ -3,4 +3,16 @@ def get_gif(genre)
+
+  def genre_top_three(genre)
+    genre_movies(genre).sort_by(&:avg_rating).reverse.first(3)
+  end
```
These feel like they should belong in a Genre model, which makes me wonder if we should've made a separate genre table...

@Koala-t @davehenton @samuelparker thoughts?

salogel42 commented on app/views/genres/show.html.erb:
```diff
@@ -0,0 +1,45 @@
+<% genre = params[:genre] %>
```
The view should not need to know about the params!  The controller should grab this and pass it through.

salogel42 commented on app/views/genres/show.html.erb:
```diff
@@ -0,0 +1,45 @@
+      </div>
+    </div>
+  </div>
+</div>
```
This is (or at least should be) exactly the same as the one on the movie index (homepage), except for how we get the list of movies.  Refactor it out into a partial that takes the movie list as a local so that there's a single source of truth and both will get any updates we make (keeping multiple copies of something in sync is a huge nightmare)!

salogel42 commented on app/views/movies/index.html.erb:
```diff
@@ -25,13 +25,15 @@
                 <iframe src=<%= get_gif(genre).to_s %>></iframe>
               </div>
               <h3 class="genres"><%= genre %></h3>
             </div>
```
Indent the section inside the link_to.

salogel42 commented on config/routes.rb:
```diff
@@ -5,6 +5,7 @@
 
   root 'movies#index'
 
+  get '/genres/:genre' => 'genres#show'
```
use resources with only

salogel42 reviewed (changes requested):
@Koala-t couple of suggestions, mainly making the card into a partial instead of copy/pasta

Koala-t committed:
refactor helper-method and delegate top-3-movie card-generation to a partial

Koala-t committed:
Add genres and genres_movies migrations and models

-remove unnecessary helper methods
-refactor movie-index and genre-show pages to use the database-genres
-add join table for associations between genres and movies
-update seed file to make the associations

samuelparker commented:
I am reviewing this now.

Commit:
resolve conflict with stylesheet
